spring-cloud-kubernetes-client-discovery/src/main/java/org/springframework/cloud/kubernetes/client/discovery/reactive/KubernetesInformerReactiveDiscoveryClientAutoConfiguration.java

```diff
@@ -84,13 +95,43 @@ public KubernetesInformerReactiveDiscoveryClient kubernetesReactiveDiscoveryClie
 				serviceLister, endpointsLister, serviceInformer, endpointsInformer, properties);
 	}
 
+	/**
+	 * Post an event so that health indicator is initialized.
+	 */
+	@Bean
+	@ConditionalOnClass(name = "org.springframework.boot.actuate.health.ReactiveHealthIndicator")
+	@ConditionalOnDiscoveryHealthIndicatorEnabled
+	KubernetesDiscoveryClientHealthIndicatorInitializer reactiveIndicatorInitializer(
+			ApplicationEventPublisher applicationEventPublisher, PodUtils<?> podUtils) {
+		LOG.debug(() -> "Will publish InstanceRegisteredEvent from reactive implementation");
+		return new KubernetesDiscoveryClientHealthIndicatorInitializer(podUtils, applicationEventPublisher);
+	}
+
+	/**
+	 * unlike the blocking implementation, we need to register the health indicator.
+	 */
+	@Bean
+	@ConditionalOnClass(name = "org.springframework.boot.actuate.health.ReactiveHealthIndicator")
+	@ConditionalOnDiscoveryHealthIndicatorEnabled
+	ReactiveDiscoveryClientHealthIndicator kubernetesReactiveDiscoveryClientHealthIndicator(
```

Comment:
it took forever to understand what is going on. It has to do with the fact that `org.springframework.cloud.client.CommonsClientAutoConfiguration` and `org.springframework.cloud.client.ReactiveCommonsClientAutoConfiguration` have a different way of working. One expects `ReactiveDiscoveryClientHealthIndicator` to be provided, and the blocking one: creates them based on the `DiscoveryClient` bean already in the context. 

This different approach means different approaches in our case too. Not obvious (not very pleasant either), but not a big deal. I can go into a lot more details if needed, just let me know. But in general, your approach that was already here makes total sense now.